includes/libraries/class-wc-eval-math.php

```diff
@@ -318,7 +318,7 @@ private static function pfx( $tokens, $vars = array() ) {
 		 */
 		private static function trigger( $msg ) {
 			self::$last_error = $msg;
-			if ( Constants::is_true( 'WP_DEBUG' ) ) {
+			if ( ! Constants::is_true( 'DOING_AJAX' ) && Constants::is_true( 'WP_DEBUG' ) ) {
```

Comment:
Not a blocker, but do we know why this was removed? What happens to the AJAX response when we echo the error?

Reply:
@ObliviousHarmony I don't know much about this change but here is the commit that introduced it: https://github.com/woocommerce/woocommerce/commit/5626497ce12d4dbcd25550aebc5389ef31179e08

Reply:
to quote Vedanshu, 

> this is because output of echo mess with JSON formatting, so you will end with invalid JSON response from ajax request

> we already add error array object in JSON response when there is an error

Reply:
I'm not entirely sure why but this was inverted in my brain and I thought the code was removed, not added!